cli/cli-utils/src/getConfig.ts

```diff
@@ -35,7 +36,19 @@ export async function getConfig (
     })
   }
   if (!config.ignorePnpmfile) {
-    const { hooks, resolvedPnpmfilePaths } = requireHooks(config.lockfileDir ?? config.dir, config)
+    const pnpmfiles = config.pnpmfile == null ? [] : Array.isArray(config.pnpmfile) ? config.pnpmfile : [config.pnpmfile]
+    if (config.configDependencies) {
+      const configModulesDir = path.join(config.lockfileDir ?? config.rootProjectManifestDir, 'node_modules/.pnpm-config')
+      for (const configDepName of Object.keys(config.configDependencies)) {
```

Comment:
`Object.keys` alone doesn't give deterministic order. You may need to add a `.sort()`.